monix-catnap/shared/src/main/scala/monix/catnap/ConcurrentQueue.scala

```diff
@@ -277,6 +277,15 @@ final class ConcurrentQueue[F[_], A] private (
     notifyProducers()
   }
 
+  /** Checks if the queue is empty.
+    *
+    * '''UNSAFE PROTOCOL:''' Concurrent shared state changes very frequently, therefore this function might yield nondeterministic results.
+    */
+  @UnsafeProtocol
+  @UnsafeBecauseImpure
```

Comment:
It is actually pure because of `F[_]`

Reply:
oh yeah, crap, you're totally right - i forgot about that fact when i was reading the description of that annotation haha